SimPlatform/GpaLib/bar.cpp

```diff
@@ -0,0 +1,52 @@
+#include "bar.h"
+#include<iostream>
+#include<vector>
+
+#include<QProgressBar>
+
+
+Bar::Bar(QWidget *parent) : QDialog (parent)
+{
+    bar = new QProgressBar;
+
+        QHBoxLayout *layout = new QHBoxLayout;
+        layout->addWidget(bar);
+
+    setLayout(layout);
+}
+
+Bar::~Bar(){
+    delete bar;
+}
+
+void Bar::setValue(int min, int max, int current)
+{
+
+    bar->setMinimum(min);
+    bar->setMaximum(max);
+    bar->setValue(current);
+
+}
+
+void Bar::setText(QString text)
+{
+    bar->setToolTip(text);
+}
+
+void Bar::updatable(bool value)
+{
+    if(value == true){
+    bar->updatesEnabled(); }
```

Comment:
Can we disable the update if the bool is false?

Reply:
Yes! the method to disable update is added in case of a bool is false. 007f7f72190ce61ab8fd5e28ed015d17bcc70f87